angularjs-portal-home/src/main/webapp/my-app/layout/controllers.js

```diff
@@ -171,6 +141,41 @@ define(['angular', 'jquery'], function(angular, $) {
       vm.init();
     }])
 
+  .controller('RemoveWidgetController', ['$scope', 'layoutService',
+    '$sessionStorage', function($scope, layoutService,
+      $sessionStorage) {
+        var vm = this;
+        /**
+         * Remove widget from home layout
+         * @param fname
+         */
+        vm.removePortlet = function removePortletFunction(fname) {
+          layoutService.removeFromHome(fname).success(function() {
+            $scope.$apply(function(request, text) {
+              var result = $.grep($scope.$parent.layout, function(e) {
+                return e.fname === fname;
+              });
+              var index = $.inArray(result[0], $scope.$parent.layout);
+              // remove
+              $scope.$parent.layout.splice(index, 1);
+              if ($sessionStorage.marketplace != null) {
+                var marketplaceEntries = $.grep($sessionStorage.marketplace,
+                  function(e) {
+                    return e.fname === result[0].fname;
+                  });
+                if (marketplaceEntries.length > 0) {
+                  marketplaceEntries[0].hasInLayout = false;
+                }
+              }
+            });
+          }).error(
+            function(request, text, error) {
+              alert('Issue deleting ' + fname +
```

Comment:
probably outside of the scope of this PR, but a [toast](https://material.angularjs.org/latest/api/service/$mdToast) would look a lot nicer than an alert.

Reply:
Agree. [Evolving in-app messaging capabilities of `uPortal-app-framework` is on MyUW's roadmap for this quarter](https://docs.google.com/document/d/1yet7w3Nz50gcH19edMCLqCxde0IcQzmVTrZwxj0jzPk/edit#heading=h.j3t43jedaugt), for whatever that's worth.